### PR TITLE
Upgrade bootstrap 5.3

### PR DESCRIPTION
**Pull request by thijskh**
This touches quite some things so no doubt a quirk here or there will be left or there are areas to improve even further.

### CODE REVIEW

File: webapp/src/Controller/Team/SubmissionController.php [hunk 1]
```diff
@@ -95,6 +95,7 @@ public function createAction(Request $request, ?Problem $problem = null): Respon
         }
 
         $data = ['form' => $form->createView(), 'problem' => $problem];
+        $data['validFilenameRegex']    = SubmissionService::FILENAME_REGEX;
```
Comment (nickygerritsen) [inline webapp/src/Controller/Team/SubmissionController.php:98]: The whitespace around the = is a bit weird, isn't it?

File: webapp/templates/base.html.twig [hunk 2]
```diff
@@ -45,16 +45,6 @@
         var domjudge_base_url = "{{ app.request.getBaseURL() }}";
     {% endif %}
     $(function () {
-        $('body').on('change', '.custom-file-input', function () {
```
Comment (nickygerritsen) [inline webapp/templates/base.html.twig:48]: Is the file widget now natively supported, or is it now different?

**Review by nickygerritsen — APPROVED** [review]
Looks good in general. I guess we just have to see where we missed stuff or what looks ugly.

**Comment by nickygerritsen** [comment]
The toggles that are autosaved after clicking them (like on the language page) don't work anymore,

**Comment by nickygerritsen** [comment]
Links are now underscored by default (without hovering), which to me looks more ugly.

**Comment by thijskh** [comment]
> Links are now underscored by default (without hovering), which to me looks more ugly.

This is indeed a new default in BS5. I think it's some idea about accessibility. I also think it's not so beautiful so I'd indeed agree to remove this.

File: webapp/templates/base.html.twig (hunk 2 above)
Comment (thijskh) [reply webapp/templates/base.html.twig:48]: The default view is i.m.o. beter and I've moved this functionality of listing all filenames to the place where it matters and solved it there in an even better way.

Comment (nickygerritsen) [reply webapp/templates/base.html.twig:48]: Yeah I just looked at it and it looks good.

**Comment by thijskh** [comment]
Added the fixes in cba0fae

**Review by nickygerritsen — APPROVED** [review]

**Pull request merged**